src/pineko/scale_variations.py

```diff
@@ -6,55 +6,149 @@
 
-    k_R is defined as the log(mu_{R}^{2}/ Q^2).
+    Parameters
+    ----------
+    m : int
+        first non zero perturbative order
+    nec_order : tuple(int)
+        tuple of the order for which the sv contribution is asked for
+    to_contruct_order : tuple(int)
+        tuple of the sv order to be constructed
+    nf : int
+        number of active flavors
+    kR : float
+        log of the ratio between renormalization scale and Q
 
-    nf is the number of active flavors.
+    Returns
+    -------
+    float
+        full contribution of ren sv
     """
-    return 1.0
-
+    delta = to_construct_order[0] - m
+    logpart = to_construct_order[2]
+    return (kR**logpart) * ren_sv_contrib(m, delta, logpart, nec_order[0] - m, nf)
 
-def NLO_ren_var(m, k_R, nf):
-    """NLO ren scale variations depending on the first non vanishing order m.
 
-    k_R is defined as the log(mu_{R}^{2}/ Q^2).
+def compute_map_of_necessary_orders(m, delta):
+    """Compute a dictionary with all the necessary order to compute in order to have the full renormalization scale variation.
 
-    nf is the number of active flavors.
+    Parameters
+    ----------
+    m : int
+        first non zero perturbative order
+    delta : int
+        difference between asked order and m
+
+    Returns
+    -------
+    dict(tuple(int))
+        description of all the needed orders
     """
-    return -k_R * m * beta.beta_qcd((2, 0), nf)
-
+    nec_orders = {}
+    for delt in range(delta):
+        nec_orders[(m + delta, 0, delt + 1, 0)] = [
+            (m + de, 0, 0, 0) for de in range(delta - delt)
+        ]
+    return nec_orders
+
+
+def create_svonly_grid(grid, order, new_order, scalefactor):
+    """Create a grid containing only the renormalization scale variations at a given order for a grid."""
+    # Retrieve parameters to create new grid
+    bin_limits = [
+        float(bin) for bin in range(grid.raw.bins() + 1)
+    ]  # +1 because I don't know
+    lumi_grid = [pineappl.lumi.LumiEntry(mylum) for mylum in grid.raw.lumi()]
+    subgrid_params = pineappl.subgrid.SubgridParams()
+    new_order = [pineappl.grid.Order(*new_order)]
+    # create new_grid with same lumi and bin_limits of the original grid but with new_order
+    new_grid = pineappl.grid.Grid.create(
+        lumi_grid, new_order, bin_limits, subgrid_params
+    )
+    # extract the relevant order to rescale from the grid for each lumi and bin
+    grid_orders = [orde.as_tuple() for orde in grid.orders()]
+    order_index = grid_orders.index(order)
+    for lumi_index in range(len(lumi_grid)):
+        for bin_index in range(grid.raw.bins()):
+            extracted_subgrid = grid.subgrid(order_index, bin_index, lumi_index)
+            extracted_subgrid.scale(scalefactor)
+            # Set this subgrid inside the new grid
+            new_grid.set_subgrid(0, bin_index, lumi_index, extracted_subgrid)
+    return new_grid
+
+
+def create_all_necessary_grids(gridpath, order, kR, nf):
```

Comment:
```suggestion
def create_grids(gridpath, order, kR, nf):
```
if you do not specify which, the only sensible meaning is "all of them". But `create` alone would sound ambiguous, so here I would keep the `_grids`